### PR TITLE
add `ConfigObject::get-/setValue<EnumType>`

### PR DESCRIPTION
**Pull request by Swiftb0y**
This allows the callsites to avoid verbose `static_cast`s (see 491b43639506e5cba2330199451c599d2fad2da2). It also makes it easier to change the machanism used when storing enums in the config easier in the future. The remaining commits are cleanups I did along the way while making use of the new overload.

### CODE REVIEW

File: src/preferences/configobject.h
Comment (daschuer) [inline src/preferences/configobject.h:182]:
```suggestion
    template <class ResultType, std::enable_if_t<!std::is_enum_v<ResultType>, bool> = true>
    ResultType getValue(const ConfigKey& key, const ResultType& default_value) const;
    QString getValue(const ConfigKey& key, const char* default_value) const;
    template<typename EnumType, std::enable_if_t<std::is_enum_v<EnumType>, bool> = true>
```

File: src/mixxxmainwindow.cpp
Comment (daschuer) [inline src/mixxxmainwindow.cpp:177]:
```suggestion
    m_toolTipsCfg = pConfig->getValue(
```

File: src/preferences/dialog/dlgprefautodj.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefautodj.cpp:21]:
```suggestion
    RequeueIgnoreCheckBox->setChecked(m_pConfig->getValue(
```

File: src/preferences/dialog/dlgprefautodj.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefautodj.cpp:147]: connected QCheckBox::stateChanged retuns int, because of three states.

File: src/preferences/dialog/dlgprefautodj.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefautodj.cpp:50]: slotToggleRandomQueue takes int.

File: src/preferences/dialog/dlgprefautodj.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefautodj.cpp:121]: slotToggleRandomQueue takes int.

File: src/preferences/dialog/dlgprefautodj.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefautodj.cpp:177]: Is conneted to QCheckBox::stateChanged which is int.

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/preferences/dialog/dlgprefautodj.cpp
Comment (Swiftb0y) [reply src/preferences/dialog/dlgprefautodj.cpp:147]: Ah yep, oversight. I thought it was bound to `QAbstractButton::clicked`.

File: src/preferences/configobject.h
Comment (Swiftb0y) [reply src/preferences/configobject.h:160]: >  The alternative would be to merge that to main, because this is not a last minute bug fix.

I'd prefer merging to main then. The error messages generated when using `enable_if` are much uglier than the ones generated by `requires` clauses.

**Comment by Swiftb0y** [comment]
I'll push a couple fixups. If you're okay with them I'll squash them and rebase onto `main` in a single step after approval. Reverting to draft as long as the history contains fixups.

File: src/preferences/configobject.h
Comment (Swiftb0y) [reply src/preferences/configobject.h:182]: explicitly opting out of enums for the default template is unnecessary when using `requires` as the template with `requires` has a higher specificity and is thus chosen over the unconstrained template. I'd prefer to not have the negative requirement, as the maintenance burden will explode in case we add even more templates.

File: src/preferences/configobject.h
Comment (daschuer) [reply src/preferences/configobject.h:160]: OK

**Review by daschuer — APPROVED** [review]
LGTM, Ready to rebase.

File: src/preferences/configobject.h
Comment (Swiftb0y) [reply src/preferences/configobject.h]: updating clang-format fixed it.

**Comment by Swiftb0y** [comment]
rebased to main, fixups are gone and it builds. merge?

**Review by Holzhaus — APPROVED** [review]

File: .pre-commit-config.yaml
Comment (Swiftb0y) [reply .pre-commit-config.yaml:94]: No it was required to resolve the messed formatting I posted about above. I didn't pull 2.4 into `main` because [thats currently broken](https://github.com/mixxxdj/mixxx/pull/11839#issuecomment-1694210108) IIUC. I just figured I'd just update it here and the merge conflict will be easy to resolve later. The Alternative would be to cherry-pick #11889 into `main`.

Comment (Holzhaus) [reply .pre-commit-config.yaml:94]: Hmm, my local test worked fine, but if we changed deps this does look weird. Since #11839 has already been merged I'd rather take care of updating the vcpkg env now rather than risking that changes in 2.4 pile up and become a pain to resolve later on.

Comment (Holzhaus) [reply .pre-commit-config.yaml:94]: https://github.com/mixxxdj/mixxx/pull/11891

Comment (Swiftb0y) [reply .pre-commit-config.yaml:94]: I agree, I just don't like messing with the buildenv as that's not my area of expertise. Thank you for taking care.

**Review by daschuer — CHANGES_REQUESTED** [review]
OK, we see now everything is green. So I suggest to remove the .pre-commit-config.yaml and we can merge this with the failing pre-commit (if https://github.com/mixxxdj/mixxx/pull/11892 does not make it before). This will avoid later conflicts and doe snot block this PR.

**Comment by Swiftb0y** [comment]
removed the pre-commit change. CI is obviously still failing due it being broken on `main`.

**Comment by Swiftb0y** [comment]
merge?

**Comment by daschuer** [comment]
Yes, thank you.

**Comment by Swiftb0y** [comment]
Thank you

**Pull request merged**
